bin/cask

```diff
@@ -324,10 +330,21 @@ def main():
             exit_error(
                 'Python 2.6 required, yours is {0.major}.{0.minor}'.format(
                     sys.version_info))
+
+
         if len(sys.argv) > 1 and sys.argv[1] == 'exec':
+            ## never returns!
             exec_command(sys.argv[2:])
-        else:
-            exec_cask(sys.argv[1:])
+
+        ## special handling for --emacs/bin as this has to happen before
+        ## everything else, and we don't want the parsing in two places.
+        if len(sys.argv) > 1:
+            if sys.argv[1] == '--emacs':
+                global EMACS
```

Comment:
> In bin/cask <https://github.com/cask/cask/pull/295#discussion_r31383712>:
>
> >              exec_command(sys.argv[2:])
> > -        else:
> > -            exec_cask(sys.argv[1:])
> > +
> > +        ## special handling for --emacs/bin as this has to happen before
> > +        ## everything else, and we don't want the parsing in two places.
> > +        if len(sys.argv) > 1:
> > +            if sys.argv[1] == '--emacs':
> > +                global EMACS
>
> Actually, cask already does that, with sys.argv and more subtly in
> get_cask_path, when the environment is used.
>

It's not about the environment, it's about the *global* mutable variable.
I just don't want to maintain code with mutable shared state, and I don't
think that it's good style in a language that has other ways.

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cask/cask/pull/295/files#r31383712>.
>